front/src/common/api/generatedEditoastApi.ts

```diff
@@ -3310,7 +3311,101 @@ export type TrainSchedule = {
   start_time: string;
   train_name: string;
 };
+export type CategoryChangeGroup = {
+  value: string;
+};
+export type ConstraintDistributionChangeGroup = {
+  value: Distribution;
+};
+export type InitialSpeedChangeGroup = {
+  value: number;
+};
+export type LabelsChangeGroup = {
+  value: string[];
+};
+export type TrainScheduleOptions = {
+  use_electrical_profiles?: boolean;
+  use_speed_limits_for_simulation?: boolean;
+};
+export type OptionsChangeGroup = {
+  value: TrainScheduleOptions;
+};
+export type Margins = {
+  boundaries: string[];
+  /** The values of the margins. Must contains one more element than the boundaries
+    Can be a percentage `X%` or a time in minutes per 100 kilometer `Xmin/100km` */
+  values: string[];
+};
+export type PathItem = PathItemLocation & {
+  /** Metadata given to mark a point as wishing to be deleted by the user.
+    It's useful for soft deleting the point (waiting to fix / remove all references)
+    If true, the train schedule is consider as invalid and must be edited */
+  deleted?: boolean;
+  id: string;
+};
+export type PowerRestrictionItem = {
+  from: string;
+  to: string;
+  value: string;
+};
+export type ScheduleItem = {
+  /** The expected arrival time at the stop.
+    This will be used to compute the final simulation time. */
+  arrival?: string | null;
+  at: string;
+  /** Whether the schedule item is locked (only for display purposes) */
+  locked?: boolean;
+  reception_signal?: ReceptionSignal;
+  /** Duration of the stop.
+    Can be `None` if the train does not stop.
+    If `None`, `reception_signal` must be `Open`.
+    `Some("PT0S")` means the train stops for 0 seconds. */
+  stop_for?: string | null;
+};
+export type PathAndScheduleChangeGroup = {
+  margins: Margins;
+  path: PathItem[];
+  power_restrictions: PowerRestrictionItem[];
+  schedule: ScheduleItem[];
+};
+export type RollingStockChangeGroup = {
+  category?: TrainCategory | null;
+  comfort: Comfort;
+  rolling_stock_name: string;
+};
+export type SpeedLimitTagChangeGroup = {
+  value?: string | null;
+};
+export type StartTimeChangeGroup = {
+  value: string;
+};
+export type TrainNameChangeGroup = {
+  value: string;
+};
+export type PacedTrainException = (
+  | {
+      type: 'created';
```

Comment:
I don't think this should be exposed in the public REST API. This is strictly internal Rust details.

Reply:
Yes, but the OpenAPI is generated from Rust types. If I remove the type, it generates weird definitions and causes some tests to fail because the enum can’t be properly inferred. One possible solution would be to avoid using an enum and instead make occurrence_id and disabled optional fields. ? @leovalais @flomonster

Reply:
> Is it having nothing for Created and just two fields disabled and occurrence_id in the Modiefied case?
it already the case, i'm proposing to not having an enum ```rust
#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq, Eq, ToSchema)]
#[serde(tag = "type", rename_all="lowercase")]
pub enum PacedTrainExceptionType {
    #[default]
    Created,
    Modified {
        occurence_index: i32,
        disabled: bool,
    },
}
```
and just optional fields (occurence_index, disabled)